### ISSUE TITLE
Experiencing compatibility issues with at least one Pleroma server

### ISSUE THREAD

**Issue by dariusk** [dariusk/rss-to-activitypub#11]
See thread, need to dig into this:

https://friend.camp/@darius/101633836271141997

**Comment by wakest** [comment]
@spazradio@bots.tinysubversions.com working for me from https://social.wake.st (mastodon) but just tried to follow from https://cathoderay.tube (pleroma) and get I get the hourglass of with "Awaiting approval. Click to cancel follow request"

**Comment by dariusk** [comment]
Thanks, that tracks with what I'm hearing. I probably won't get to this until March 1 (I'm traveling) but I do have a test pleroma server up and running so I can track this down.

**Comment by dariusk** [comment]
Okay I've done some testing. __This is using Pleroma v0.9.99__. Here is my best documentation of what is going on:

* From my test pleroma server, I search an account like `@pleroma_develop_branch@bots.tinysubversions.com`
* the search works fine and the account appears with the option to follow

<img width="629" alt="image" src="https://user-images.githubusercontent.com/266454/54097143-07ecd000-436c-11e9-8d79-c86dad00653f.png">

* when I hit the "Follow" button it flashes `Requesting...`, `Unfollow`, and then `Follow` in rapid succession

Behind the scenes, this is logged on my Pleroma server:

```
Mar 11 02:38:33 li1081-165 mix[5322]: 02:38:33.180 request_id=2m5ce55bt8u4ffduuc0048oh [info] POST /api/friendships/create.json
Mar 11 02:38:33 li1081-165 mix[5322]: 02:38:33.187 [info] Relaying https://pl.tinysubversions.com/activities/5a85de32-daa3-4485-8632-25b2cc21284a out
Mar 11 02:38:33 li1081-165 mix[5322]: 02:38:33.187 [info] Sending https://pl.tinysubversions.com/activities/5a85de32-daa3-4485-8632-25b2cc21284a out via AP
Mar 11 02:38:33 li1081-165 mix[5322]: 02:38:33.188 [info] Federating https://pl.tinysubversions.com/activities/5a85de32-daa3-4485-8632-25b2cc21284a to https://bots.tinysubversions.com/api/inbox
Mar 11 02:38:33 li1081-165 mix[5322]: 02:38:33.288 request_id=2m5ce55icl6ob144dc0048p1 [info] POST /users/darius/inbox
Mar 11 02:38:33 li1081-165 mix[5322]: 02:38:33.291 request_id=2m5ce55icl6ob144dc0048p1 [info] Fetching object https://bots.tinysubversions.com/u/pleroma_develop_branch via AP
Mar 11 02:38:33 li1081-165 mix[5322]: 02:38:33.298 request_id=2m5ce55icl6ob144dc0048p1 [info] Sent 200 in 9ms
Mar 11 02:38:33 li1081-165 mix[5322]: 02:38:33.298 [info] Handling incoming AP activity
Mar 11 02:38:33 li1081-165 mix[5322]: 02:38:33.689 request_id=2m5ce55bt8u4ffduuc0048oh [info] Sent 200 in 509ms
Mar 11 02:38:34 li1081-165 mix[5322]: 02:38:34.230 request_id=2m5ce57ag00j4dhvvc0048ph [info] GET /api/users/show.json
Mar 11 02:38:34 li1081-165 mix[5322]: 02:38:34.232 request_id=2m5ce57ag00j4dhvvc0048ph [info] Sent 200 in 2ms
Mar 11 02:38:34 li1081-165 mix[5322]: 02:38:34.717 request_id=2m5ce587hkbvns99gk0048q1 [info] GET /api/qvitter/statuses/notifications.json
Mar 11 02:38:34 li1081-165 mix[5322]: 02:38:34.719 request_id=2m5ce587hkbvns99gk0048q1 [info] Sent 200 in 2ms
```
You can see I sent out what is presumably the `Follow` to the inbox of the RSS bot, and shortly thereafter received a POST to `/users/darius/inbox`, which is indeed the inbox of my user on my test instance. I presume this is the `Accept` event being sent back by my RSS-to-AP server. If I go take a look there, what was sent is:

```js
{ '@context': 'https://www.w3.org/ns/activitystreams',
id:
 'https://bots.tinysubversions.com/28d2d0afe817fa6fed275e8cce3421fe',
type: 'Accept',
actor: 'https://bots.tinysubversions.com/u/pleroma_develop_branch',
object:
 { '@context':
    [ 'https://www.w3.org/ns/activitystreams',
      'https://pl.tinysubversions.com/schemas/litepub-0.1.jsonld' ],
   actor: 'https://pl.tinysubversions.com/users/darius',
   cc: [ 'https://www.w3.org/ns/activitystreams#Public' ],
   context:
    'https://pl.tinysubversions.com/contexts/13023db2-98ed-4d3b-9982-744f6cd868b8',
   context_id: 41,
   id:
    'https://pl.tinysubversions.com/activities/f186ba1e-a5b3-4a87-b6b3-07f13657818c',
   object: 'https://bots.tinysubversions.com/u/pleroma_develop_branch',
   published: '2019-03-11T02:43:59.841940Z',
   state: 'pending',
   to: [ 'https://bots.tinysubversions.com/u/pleroma_develop_branch' ],
   type: 'Follow' } }
```

[Per the AP spec](https://www.w3.org/TR/activitystreams-vocabulary/#dfn-accept) this seems about right to me (the contents of `object` is literally just the object the RSS server received from Pleroma on the Follow).

I would assume the Follow request from Pleroma would resolve at this point, but it's not. The account is not added to the FollowersCollection on Pleroma:

<img width="660" alt="image" src="https://user-images.githubusercontent.com/266454/54097551-e55bb680-436d-11e9-868a-c617aea91894.png">

And strangely, if I reload the search page it gives me the "unfollow" option for all the bots I've been testing with, even though they are not followed by my user as far as I can tell.

<img width="685" alt="image" src="https://user-images.githubusercontent.com/266454/54097574-0a502980-436e-11e9-8ede-492635a88499.png">

Any thoughts on what is going on here?

**Comment by kaniini** [comment]
Pleroma does not like your actors because they do not declare a followers collection.  This is, presumably, allowed per AP, but Pleroma requires the declaration because we need to know what URIs need special handling (such as the followers collection URI).
